### PR TITLE
Upgrade of blade.py

### PR DESCRIPTION
**Pull request by ggurioli**
An upgrade of the blade.py to handle the porting from pythonocc 0.18.x to pythonocc 7.4.0 is considered. Two new methods (in light of a future use of BladeX for the whole propeller generation) are included: generate_root (to close the blade surface at the root); blade_solid (top assemble upper/lower faces, tip and root in a solid). An alternative method to export lower/upper faces, tip and root of the generated blade in format .stl (generate_stl) has been included, in order to get rid of the use of OCC.SMESH.

### CODE REVIEW

File: bladex/blade.py [hunk 1]
```diff
@@ -788,32 +787,95 @@ def generate_iges(self,
...
+        sewer.Add(self.generated_upper_face)
+        sewer.Add(self.generated_lower_face)
+        sewer.Add(self.generated_tip)
+        sewer.Add(self.generated_root)
+        sewer.Perform()
+        result_shell = sewer.SewedShape()
+        print("Number of sewed shape free edges: ",sewer.NbFreeEdges())
+        solid_maker = BRepBuilderAPI_MakeSolid()
+        solid_maker.Add(OCC.Core.TopoDS.topods_Shell(result_shell))
+        print("Solid done?",solid_maker.IsDone())
```
Comment (mtezzele) [inline bladex/blade.py:859]: Here better to change with a check on solid_maker.IsDone(), and if not done raise an error

File: bladex/blade.py [hunk 2]
```diff
@@ -626,12 +565,58 @@ def _generate_tip(self, maxDeg):
         # Returns the Face generated by each edge of the first section
         self.generated_tip = generator.GeneratedFace(edge_1)
 
+    def _generate_root(self, maxDeg):
```
Comment (mtezzele) [inline bladex/blade.py:568]: Please no camel for the arguments: maxDeg -> max_deg.
Refactor all the code please

File: bladex/blade.py [hunk 3]
```diff
@@ -9,68 +9,46 @@
 class Blade(object):
     """
     Bottom-up parametrized blade construction.
-
```
Comment (ndem0) [inline bladex/blade.py:12]: Are you sure that (some of) the blank lines are not required for properly generate documentation?
In any case, it is not very useful changing 30 lines of code (actually docstrings) just for avoiding blank lines

File: bladex/blade.py [hunk 4]
```diff
@@ -788,32 +787,95 @@ def generate_iges(self,
...
+            start_display()
+            
+        sewer = BRepBuilderAPI_Sewing(1e-2)
+        sewer.Add(self.generated_upper_face)
+        sewer.Add(self.generated_lower_face)
+        sewer.Add(self.generated_tip)
+        sewer.Add(self.generated_root)
+        sewer.Perform()
+        result_shell = sewer.SewedShape()
+        print("Number of sewed shape free edges: ",sewer.NbFreeEdges())
```
[the remark this answers is not in the document]
Comment (ggurioli) [reply bladex/blade.py:856]: Ok, thanks.

File: bladex/blade.py (hunk 1 above)
Comment (ggurioli) [reply bladex/blade.py:859]: Thank you for the suggestion. Is it fine to use "raise Exception"?

File: bladex/blade.py (hunk 2 above)
Comment (ggurioli) [reply bladex/blade.py:568]: That's fine, I will replace maxDeg by max_deg throughout the whole code.

File: bladex/blade.py (hunk 3 above)
Comment (ggurioli) [reply bladex/blade.py:12]: Thank you for pointing it out. I'm going to reintroduce the original blank lines.

File: bladex/blade.py [hunk 5]
```diff
@@ -853,10 +922,10 @@ def generate_blade_solid(self,
         sewer.Add(self.generated_root)
         sewer.Perform()
         result_shell = sewer.SewedShape()
-        print("Number of sewed shape free edges: ",sewer.NbFreeEdges())
         solid_maker = BRepBuilderAPI_MakeSolid()
         solid_maker.Add(OCC.Core.TopoDS.topods_Shell(result_shell))
-        print("Solid done?",solid_maker.IsDone())
+        if not solid_maker.IsDone():
+            raise RuntimeError('Solid maker undone')
```
Comment (mtezzele) [inline bladex/blade.py:928]: maybe instead of undone, which could be misleading, we can be more verbose.

File: tests/test_blade.py [hunk 6]
```diff
@@ -5,7 +5,8 @@
 import os
 import matplotlib.pyplot as plt
 from mpl_toolkits.mplot3d import Axes3D
-from OCC.TopoDS import TopoDS_Shape
+from OCC.Core.TopoDS import TopoDS_Shape, TopoDS_Solid
+from OCC.Extend.DataExchange import write_stl_file
```
Comment (mtezzele) [inline tests/test_blade.py:9]: I think you can remove this line

Comment (ggurioli) [reply tests/test_blade.py:9]: Yes, I have checked and you are right: the write_stl_file import is not needed. I will remove it, thanks.

**Pull request merged**